kernel/packages/shared/atlas/sagas.ts

```diff
@@ -64,130 +56,125 @@ function* fetchTiles() {
-  const data = atlasState.marketName
-  const targets: Record<string, MarketEntry> = {}
+  // renew atlas state to get loaded scenes information
+  atlasState = yield select(state => state.atlas)
 
-  Object.keys(data).forEach(index => {
-    const parcel = data[index]
-    if (atlasState.alreadyReported[`${parcel.x},${parcel.y}`]) {
-      return
-    }
-    if (Math.abs(parcel.x - parcelCoords.x) > maxScenesAround) {
-      return
-    }
-    if (Math.abs(parcel.y - parcelCoords.y) > maxScenesAround) {
-      return
-    }
-    targets[index] = parcel
-  })
-  return targets
+  yield call(reportScenes, atlasState, [...sceneIdsSet])
 }
 
-export function* reportScenes(marketplaceInfo?: AtlasState, selection?: Record<string, MarketEntry>): any {
-  const atlasState = marketplaceInfo ? marketplaceInfo : ((yield select(state => state.atlas)) as AtlasState)
-  const data = selection ? selection : atlasState.marketName
-  const mapByTypeAndName: Record<string, { x: number; y: number }[]> = {}
-  const typeAndNameKeys: string[] = []
-  const keyToTypeAndName: Record<string, { type: number; name: string }> = {}
-  Object.keys(data).forEach(index => {
-    const parcel = data[index]
-    const name = getNameFromAtlasState(atlasState, parcel.x, parcel.y)
-    const type = getTypeFromAtlasState(atlasState, parcel.x, parcel.y)
-    const key = `${type}_${name}`
-    if (!mapByTypeAndName[key]) {
-      mapByTypeAndName[key] = []
-      typeAndNameKeys.push(key)
-      keyToTypeAndName[key] = { type, name }
-    }
-    mapByTypeAndName[key].push({ x: parcel.x, y: parcel.y })
+export function* reportScenes(atlas?: AtlasState, sceneIds: string[] = []): any {
+  if (!atlas) {
+    return
+  }
+
+  const scenes = sceneIds.map(sceneId => atlas.idToScene[sceneId])
+
+  const minimapSceneInfoResult: MinimapSceneInfo[] = []
+
+  scenes.forEach(scene => {
+    const parcels: Vector2Component[] = []
+    let isPOI: boolean = false
+
+    scene.sceneJsonData?.scene.parcels.forEach(parcel => {
+      let xyStr = parcel.split(',')
+      let xy: Vector2Component = { x: parseInt(xyStr[0], 10), y: parseInt(xyStr[1], 10) }
+
+      if (CAMPAIGN_PARCEL_SEQUENCE.some(poi => poi.x === xy.x && poi.y === xy.y)) {
+        isPOI = true
+      }
+
+      parcels.push(xy)
+    })
+
+    minimapSceneInfoResult.push({
+      owner: getOwnerNameFromJsonData(scene.sceneJsonData),
+      description: getSceneDescriptionFromJsonData(scene.sceneJsonData),
+      previewImageUrl: scene.sceneJsonData?.display?.navmapThumbnail,
+      name: scene.name,
+      type: scene.type,
+      parcels,
+      isPOI
+    })
   })
-  window.unityInterface.UpdateMinimapSceneInformation(
-    typeAndNameKeys.map(key => ({
-      name: keyToTypeAndName[key].name,
-      type: keyToTypeAndName[key].type,
-      parcels: mapByTypeAndName[key]
-    }))
-  )
+
+  defaultLogger.info(`minimap`, minimapSceneInfoResult)
```

Comment:
Remember to remove the logs (or put them under a disabled constant)

Reply:
Sure enough. I will remove them, just added them for debugging during development.